### PR TITLE
Changes to pass on the OS version details. This will allow us to have this data in Redshift.

### PR DESCRIPTION
**Pull request by dariuskilstein**
So, I have no idea how to test this locally.

Last time I deployed ua-parser, I just pushed it live and crossed my fingers.......

I've obviously tested the regex at https://regex101.com/

Thoughts?

### CODE REVIEW

**Comment by joekearney** [comment]
Ooh well done.

Good start. I wonder if we should take inspiration from the [forked repo](https://github.com/tobie/ua-parser/blob/master/regexes.yaml#L550-L584). I also wonder if we should get up to date with that, but that seems like a much bigger effort...

**Comment by dariuskilstein** [comment]
can do....I don't think it will be that difficult. They only have like 2mths worth of changes to the file ;-0

I can just diff and throw it in....

**Comment by dariuskilstein** [comment]
OK @joekearney that is now done in a new commit.

I went through each commit they had individually and added them in :-)

I only updated the regexes.yaml file as I guess that's all we care about.

Hope that helps.

File: test_resources/test_user_agent_parser_os.yaml [hunk 1]
```diff
@@ -2029,3 +2029,10 @@ test_cases:
     patch: '2'
     patch_minor:
 
+
+  - user_agent_string: 'SoundCloud-Android/14.08.21-17 (Android 4.4.2; samsung SAMSUNG-SGH-I337)'
```
Comment (julioz) [inline test_resources/test_user_agent_parser_os.yaml:2033]: This was changed [here](https://github.com/soundcloud/android-listeners/pull/7706) recently. We are sending the following user agent:
`SoundCloud/2017.04.18-release (Android 4.1.1; Samsung GT-I9082)`

Notice how the first part (the tag) is not `SoundCloud-Android` anymore, but instead, just `SoundCloud`.
I've added one test to verify it on #5.

**Review by julioz — CHANGES_REQUESTED** [review]

File: regexes.yaml [hunk 2]
```diff
@@ -10,6 +10,9 @@ user_agent_parsers:
   - regex: '(SoundCloud-Android)/(\d+)\.(\d+)\.(\d+)'
     family_replacement: 'SoundCloud'
 
+  - regex: '(Android) (\d+)\.(\d+)\.(\d+)'
```
Comment (joekearney) [inline regexes.yaml:13]: @dariuskilstein should this now be lower, with the other android section? Is this regex in fact covered by the ones below, e.g. `'Android[\- ][\d]+\.[\d]+\.[\d]+; ([A-Za-z0-9 _-]+) '`?

**Comment by dariuskilstein** [comment]
Hey!

Thanks for the input fellas.

@julioz I can confirm that the current file will work for the new case. We have both "Soundcloud-Android" and just "Soundcloud" cases covered (the 2nd and 3rd regex's in the file).

@joekearney those android things at the bottom don't cover it. Actually, the 4th regex in the list (the one I added), covers it. 

With regards to "where these things should go in the file", I've made an assumption on how this ua-parser library works, which is:

There are three cases to take data out of this: 1/ OS data, 2/ Device data, 3/ Family data.
Hence why we see things like "os_replacement", "device_replacement", "family_replacement".

As far as I can tell, the code looks for the first regex match of the specific kind. i.e. If it is looking for OS info, it looks for the first "os_replacement" match.

Hence we'd want "'(Android) (\d+)\.(\d+)\.(\d+)'" to be close to the top.


I could be wrong.....it'd be great to be able to test this but I'm not really sure how :/

**Comment by dariuskilstein** [comment]
@joekearney 

We good to merge this?

TBH it's been a long time since I've deployed eventlogger-pipeline so it'd be great to have some help...or perhaps this can be included in the next time your team deploys?

**Comment by joekearney** [comment]
I'm going to close this, as it doesn't seem to warrant high priority. I've added a ticket [here](https://soundcloud.atlassian.net/browse/DATA-1360) to figure out how to do all of this better, but I'm not expecting it to get prioritised unless it's important.

Please reopen @dariuskilstein @julioz if you think we should reconsider this.

**Pull request closed without merging**